src/System.Net.Sockets/tests/FunctionalTests/Connect.cs

```diff
@@ -85,6 +85,82 @@ public async Task Connect_AfterDisconnect_Fails()
                 }
             }
         }
+
+        [Fact]
+        [PlatformSpecific(~TestPlatforms.OSX)] // Not supported on OSX.
+        public async Task ConnectGetsCanceledByDispose()
+        {
+            bool usesApm = UsesApm ||
+                           (this is ConnectTask); // .NET Core ConnectAsync Task API is implemented using Apm
+
+            // We try this a couple of times to deal with a timing race: if the Dispose happens
+            // before the operation is started, we won't see a SocketException.
+
+            SocketError? localSocketError = null;
+            bool disposedException = false;
+            for (int i = 0; i < 10 && !localSocketError.HasValue; i++)
+            {
+                var client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+
+                Task connectTask = Task.Factory.StartNew(() =>
+                {
+                    ConnectAsync(client, new IPEndPoint(IPAddress.Parse("1.1.1.1"), 23)).GetAwaiter().GetResult();
+                }, TaskCreationOptions.LongRunning);
```

Comment:
Why is this spinning up a new thread only to then block it by creating an asynchronous operation and synchronously waiting for it?

Reply:
I was a bit too paranoid about blocking the threadpool when writing these tests.